nima_core/storage/unified_db.py

```
@@ -423,6 +423,15 @@ def get_unified_conn(db_path: str | Path | None = None) -> Any:
                 # that recreate the DB file must call
                 # reset_schema_init_cache().
                 _SCHEMA_INITIALIZED_PATHS.add(path_key)
+    
+    # Register NIMA custom SQL functions (json_changed_fields) on every
+    # connection so enrichment/NER code that uses get_unified_conn has them.
+    from nima_core.storage.open import _json_changed_fields
+    try:
+        conn.create_function("json_changed_fields", 2, _json_changed_fields)
+    except sqlite3.Error as exc:
```

Ruff (0.15.10)

[error] Undefined name `sqlite3`

(F821)